### ISSUE TITLE
CSV friendlier SP results

### ISSUE THREAD

**Issue by dnc-msimons** [BrentOzarULTD/SQL-Server-First-Responder-Kit#388]
The results of the stored procedures are great in SSMS, but exporting them to CSV to distribute to others as documentation or evidence for change management makes for a mess.   Exporting the results of the stored procedures to CSV from SSMS produces some pretty scary/hard-to-read CSV with inconsistent columns due to the use of commas and the lack of column headers in the export.  

1. SSMS doesn't seem to quote the individual field values which causes Excel to break any embedded commas into multiple columns which causes jagged and inconsistent data spread across columns between rows.   Maybe consider removing use of commas in output or maybe quoting the field values?

2. SSMS doesn't seem to export column names/headers to the CSV export either (is there a setting to change this?)  which makes for lots of data that doesn't correlate easily, especially when dealing with comma issue above.  Maybe an option for a "header result row" at the top using sorting -99 or something to that effect, similar to -1 versioning row.

Can provide samples or steps to reproduce if the above is not very clear.

**Comment by BrentOzar** [comment]
Lemme take a step back - what are you doing with the output? Can you describe your process? Making sure we're solving the right problem. Tell me about why you want SQL to format the results in a CSV. Thanks!

---
Tiny glass keyboard
Typos flowing like rivers
As winter snow thaws

**Comment by dnc-msimons** [comment]
TLDR: right-clicking on the Results window in SSMS and selecting "Save Results as" to a CSV file, makes for a pretty uncontextable (new word) blob of text with inconsistent columns.

Long version:
About half of my 200+ servers live in a heavily locked down environment due to PCI requirements (so I'm limited on how I can remove data from the environment), all running various different applications/systems owned/maintained by various different vendors.  The vendors control the applications, I just keep the DB lights on so to speak.  But I'm occasionally tasked with "Hey Mike, tell us why this server is slow/under performing."  So usually I sit in an advisory role to the vendors, but also like/need to have backing (change management reqs, etc) for any server level changes under my control before fiddling with settings.  So after running all of your queries, and analyzing them with a DBA-brain looking for where determine the right avenue to attack, I have to "present my case", so to speak, to the management/application vendors.   It would be nice to provide the results of the stored procedure (or parts of the results at least), depending on audience, as backing/evidence for what changes I am proposing.  I realize that the stored procedures aren't necessarily meant for general/lay-person consumption, but sometimes my audience is developers who should be able make sense of the information provided.   Many times, the server in question is revenue impacting and speed is of the essence, where quick and dirty facts gathering trump pretty formatting efforts, hence CSV output of result.    

As an example of the output, on a current server, sp_blitz_index results in almost 5000! findings ranging in priority from 10 to 250.   When output to CSV, due to comma and SSMS's lack of properly escaping cell values when writing CSV files, some rows spit partial data into column as far out as AT (in Excel) while other rows coming from the same result column have partial data in column W.  With so many commas used in various fields, pretty dynamically in that SP, the data is all over the map.  Even a user with relatively high Excel skills would still be hamstrung.  A different separator value than the comma might alleviate some of the strain/inconsistency.

Hopefully that helps shed light or at least where I'm coming from.

**Comment by dnc-msimons** [comment]
I guess it boils down to my expectation that if a sp/query produces say 15 columns of data, exporting it to CSV, should result in 15 columns of output, where everything breaks at the same place between columns as it did in the Results window.  MS is notoriously bad at doing this right with commas and CSV, and while this firmly falls under "MS's fault", sometimes it's easier to change our own behavior than waiting on them to change.

**Comment by BrentOzar** [comment]
Yeah, I gotta be honest - that isn't going to be doable inside the stored proc as just a suggestion. Even if you wanted to contribute the code, this is something we'd have to maintain for every new check going forward, and that's not really fair to ask of the community.

I think you're going to be better off using the output table parameters, and then using your preferred method of getting data out of tables and into CSV.

**Comment by dnc-msimons** [comment]
Thanks, figured it didn't hurt to ask.  I'm not married to CSV, per say, but it's usually a good 80% option.
